internal/adapter/relation.go

```diff
@@ -0,0 +1,87 @@
+package adapter
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/goto/shield/core/group"
+	"github.com/goto/shield/core/relation"
+	"github.com/goto/shield/core/user"
+	"github.com/goto/shield/internal/schema"
+	"github.com/goto/shield/pkg/uuid"
+)
+
+type Relation struct {
+	groupService    *group.Service
+	userService     *user.Service
+	relationService *relation.Service
+}
+
+func NewRelation(
+	groupService *group.Service,
+	userService *user.Service,
+	relationService *relation.Service,
+) *Relation {
+	return &Relation{
+		groupService:    groupService,
+		userService:     userService,
+		relationService: relationService,
+	}
+}
+
+func (a Relation) TransformRelation(ctx context.Context, rlt relation.RelationV2) (relation.RelationV2, error) {
+	rel := rlt
+
+	// If Principal is a user, then we will get ID for that user as Subject.ID
+	if rel.Subject.Namespace == schema.UserPrincipal || rel.Subject.Namespace == "user" {
+		fetchedUser, err := a.userService.GetByEmail(ctx, rel.Subject.ID)
```

Comment:
could there be a use case where we have to allow user uuid and email interchangebly? @AkarshSatija @mabdh 

if yes, let's add it right away?